### ISSUE TITLE
Add rule to enforce links as reference links

### ISSUE THREAD

**Issue by mondeja** [DavidAnson/markdownlint#321]
As default deactivated. If active, raise error here:

```markdown
I'm a paragraph with [a link](http://link-to-somewhere)
```

...but passes this:

```markdown
I'm a paragraph with [a link][link-to-somewhere]

[link-to-somewhere]: http://link-to-somewhere
```

**Comment by waldyrious** [comment]
It would also be nice if this was configurable by the length of the URL — i.e. non-reference links shorter than N characters would not trigger this rule, with the default max length being 0.

**Comment by DavidAnson** [comment]
Agree, I like the rule and the parameter suggestion.

**Comment by leetrout** [comment]
@DavidAnson @mondeja Any interest in an additional rule to enforce correct style on reference links for at least one unit of whitespace after the colon?


This would fail:

```diff
[link-to-somewhere]:http://link-to-somewhere
```

This would pass:

```diff
[link-to-somewhere]: http://link-to-somewhere
```

**Comment by DavidAnson** [comment]
It doesn't seem required by the specification, but I agree that one space is stylistically preferred. This would be a reasonable rule.

**Comment by leetrout** [comment]
[CommonMark deviates from this](https://spec.commonmark.org/0.29/#link-reference-definition) and says it is optional but it is mentioned explicitly in the original [Daring Fireball](https://daringfireball.net/projects/markdown/syntax#link) docs:

> Then, anywhere in the document, you define your link label like this, on a line by itself:
>
> `[id]: http://example.com/  "Optional Title Here"`
>
> That is:
>
> Square brackets containing the link identifier (optionally indented from the left margin using up to three spaces);
> followed by a colon;
> followed by one or more spaces (or tabs);
> followed by the URL for the link;
> optionally followed by a title attribute for the link, enclosed in double or single quotes, or enclosed in parentheses.

**Comment by leetrout** [comment]
Opened https://github.com/DavidAnson/markdownlint/issues/326

Sorry to interrupt this, but thanks for the quick answers!

**Comment by wolf99** [comment]
I would like this as an optional rule, either with or without the space.
(Though I prefer with the space)

**Comment by wolf99** [comment]
Possible duplicate of #40 ?

**Comment by leetrout** [comment]
Yes, looks like #40 covers the same functionality with more options.

**Comment by DavidAnson** [comment]
Agree, thanks!
